### PR TITLE
Add requestId to queries so that abandoned ones are cancelled

### PR DESCRIPTION
**Pull request by nvartolomei**

### CODE REVIEW

**Comment by nvartolomei** [comment]
I'm not able to run tests locally even on master branch btw.

**Comment by hagen1778** [comment]
Hi @nvartolomei !
Thank you for contribution!

Sure, I need to upgrade README with contributing instructions. Thanks for pointing that.

As for `requestID`.. how we can benefit from it? AFAIK, ClickHouse won't cancel a query even if request was cancelled. This will only help if you have [chproxy](https://github.com/Vertamedia/chproxy) instance in the middle.

**Comment by nvartolomei** [comment]
There are multiple benefits:

1. If a huge chunk of data was requested by accident, a cancelled request will free network bandwidth between Grafana server and the client, also will prevent browser to parse huge amounts of data when it is not needed (Grafana UI frequently freezes the browser).
2. Will free resources on Grafana server.
3. If used together with chproxy, ClickHouse request will be cancelled. (at the same time I'm trying to get this in https://github.com/yandex/ClickHouse/pull/4213)

**Comment by hagen1778** [comment]
This [yandex/ClickHouse#4213](https://github.com/yandex/ClickHouse/pull/4213) is great initiative!

File: src/datasource.ts [hunk 1]
```diff
@@ -79,23 +80,10 @@ export class ClickHouseDatasource {
...
-                queries.push(q);
-                try {
-                    let queryAST = new Scanner(q).toAST();
-                    keyColumns.push(queryAST['group by'] || []);
-                } catch (err) {
-                    console.log('AST parser error: ', err)
-                }
-            }
-        });
+        const queries = _(options.targets)
```
Comment (simPod) [inline src/datasource.ts]: What does this do please? Why is there `_`? Thanks!

Comment (simPod) [reply src/datasource.ts]: Lodash chaining, got it. nvm

**Pull request merged**